packages/apps/tools/src/components/Global/ModuleExplorer/SidePanel/outline.tsx

```diff
@@ -1,14 +1,100 @@
-import { Heading, Text } from '@kadena/react-ui';
+import { contractParser } from '@kadena/pactjs-generator';
+import type { ITreeProps } from '@kadena/react-ui';
+import { Button, Heading, Text, Tree } from '@kadena/react-ui';
 
+import type { IChainModule } from '../types';
+
+import useTranslation from 'next-translate/useTranslation';
 import React from 'react';
 
-interface IOutlineProps extends React.HTMLAttributes<HTMLDivElement> {}
+export interface IOutlineProps extends React.HTMLAttributes<HTMLDivElement> {
+  selectedModule?: IChainModule;
+  onInterfaceClick: (module: IChainModule) => void;
+}
+
+type Contract = ReturnType<typeof contractParser>[0][0]; // TODO: Should we improve this because it's a bit hacky?
+
+const contractToTreeItems = (
```

Comment:
Just a thought, maybe it would be better if this is in another file (utils, for example) not to pollute this file too much

Reply:
Agreed, will do that, thanks

Reply:
Done!